src/main/java/com/currencyexchange/cache/ExchangeRateCacheService.java

```diff
@@ -53,4 +53,18 @@ public Map<String, BigDecimal> getExchangeRates(String currency) {
     throw new RateNotFoundInCacheException(
         "Exchange rates for currency " + currency + " not found in cache");
   }
+
+  /**
+   * Updates the cache with the provided exchange rates.
+   *
+   * @param allRates A map where the key is the base currency, and the value is a map of target
+   *     currencies with their exchange rates.
+   */
+  public void saveRatesToCache(Map<String, Map<String, BigDecimal>> allRates) {
```

Comment:
Rename allRates to rates. 
Also rename saveRatesToCache to save - it`s cash service and we get rates param, so, It is clear what and where is saved